airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py

```diff
@@ -1,31 +1,267 @@
-from source_snowflake.schema_builder import format_field
-from unittest.mock import MagicMock
+import json
+import unittest
+from datetime import datetime
+
+from source_snowflake.schema_builder import format_field, get_generic_type_from_schema_type, SchemaTypes, \
+    convert_time_zone_time_stamp_suffix_to_offset_hours, convert_utc_to_time_zone, convert_utc_to_time_zone_date
 import pytest
 
 
-def test_format_field_semi_structured_type():
-    pass
+def test_get_generic_type_from_schema_type():
+    assert get_generic_type_from_schema_type(SchemaTypes.timestamp_with_timezone) == "timestamp_with_timezone"
+    assert get_generic_type_from_schema_type(SchemaTypes.timestamp_without_timezone) == "date"
+    assert get_generic_type_from_schema_type(SchemaTypes.date) == "date"
+    assert get_generic_type_from_schema_type(SchemaTypes.time_without_timezone) == "date"
+    assert get_generic_type_from_schema_type(SchemaTypes.number) == "number"
+    assert get_generic_type_from_schema_type(SchemaTypes.integer) == "number"
+    assert get_generic_type_from_schema_type(SchemaTypes.boolean) == "number"
+    assert get_generic_type_from_schema_type(SchemaTypes.string) == "string"
+    assert get_generic_type_from_schema_type(SchemaTypes.array_with_any) == "string"
+    assert get_generic_type_from_schema_type(SchemaTypes.object) == "string"
+
+
+def test_convert_time_zone_time_stamp_suffix_to_offset_hours():
+    assert convert_time_zone_time_stamp_suffix_to_offset_hours("720") == -12
+    assert convert_time_zone_time_stamp_suffix_to_offset_hours("-120") == -26
+    assert convert_time_zone_time_stamp_suffix_to_offset_hours("1560") == 2
+    assert convert_time_zone_time_stamp_suffix_to_offset_hours("1440") == 0
+    assert convert_time_zone_time_stamp_suffix_to_offset_hours("1080") == -6
+
+    # Test case for invalid input (non-numeric)
+    with pytest.raises(ValueError):
+        convert_time_zone_time_stamp_suffix_to_offset_hours("abc")
+
+
+def test_convert_utc_to_time_zone():
```

Comment:
Can we have those tests be parametrized or split into multiple test? I fear that if one fails, we won't get visibility on all the others

Reply:
I don't understand what you mean by parametrized. I defined the test as unit that verifies that the whole function is working not that a fraction of it is working. So, if something breaks early, you will  have fix the first issue as the test will stop at that point. Then when you rerun the test you will get next errors if there are errors.

Reply:
Done